daemon uses deprecated functions from Curl

```
In file included from Unvanquished/daemon/src/engine/client/dl_main.cpp:45:
Unvanquished/daemon/src/engine/client/dl_main.cpp: In member function ‘bool {anonymous}::CurlDownload::SetOptions(Str::StringRef)’:
Unvanquished/daemon/src/engine/client/dl_main.cpp:175:25: warning: ‘CURLOPT_PROTOCOLS’ is deprecated: since 7.85.0. Use CURLOPT_PROTOCOLS_STR [-Wdeprecated-declarations]
  175 |                 SETOPT( CURLOPT_PROTOCOLS, long(CURLPROTO_HTTP) )
      |                         ^~~~~~~~~~~~~~~~~
Unvanquished/daemon/src/engine/client/dl_main.cpp:175:17: note: in expansion of macro ‘SETOPT’
  175 |                 SETOPT( CURLOPT_PROTOCOLS, long(CURLPROTO_HTTP) )
      |                 ^~~~~~
/usr/include/x86_64-linux-gnu/curl/curl.h:1749:3: note: declared here
 1749 |   CURLOPTDEPRECATED(CURLOPT_PROTOCOLS, CURLOPTTYPE_LONG, 181,
      |   ^~~~~~~~~~~~~~~~~
```

Curl version from debian stable: libcurl4-gnutls-dev 7.88.1-10

--- a/src/engine/client/dl_main.cpp
+++ b/src/engine/client/dl_main.cpp
@@ -172,7 +172,11 @@ if (curl_easy_setopt(request_, option, value) != CURLE_OK) { \
 		SETOPT( CURLOPT_USERAGENT, Str::Format( "%s %s", PRODUCT_NAME "/" PRODUCT_VERSION, curl_version() ).c_str() )
 		SETOPT( CURLOPT_REFERER, Str::Format("%s%s", URI_SCHEME, Cvar::GetValue("cl_currentServerIP")).c_str() )
 		SETOPT( CURLOPT_URL, url.c_str() )
+#if CURL_AT_LEAST_VERSION(7, 85, 0)
+		SETOPT( CURLOPT_PROTOCOLS_STR, "http" )
+#else
 		SETOPT( CURLOPT_PROTOCOLS, long(CURLPROTO_HTTP) )
+#endif
 		SETOPT( CURLOPT_WRITEFUNCTION, curl_write_callback(LibcurlWriteCallback) )
 		SETOPT( CURLOPT_WRITEDATA, static_cast<void*>(this) )
 		SETOPT( CURLOPT_FAILONERROR, 1L )

